src/helpers/figures/charts/chart_common_bar_combo.ts

```diff
@@ -0,0 +1,261 @@
+import { AbstractChart } from "./abstract_chart";
+import {
+  ChartColors,
+  chartFontColor,
+  checkDataset,
+  checkLabelRange,
+  copyDataSetsWithNewSheetId,
+  copyLabelRangeWithNewSheetId,
+  createDataSets,
+  transformChartDefinitionWithDataSetsWithZone,
+  updateChartRangesWithDataSets,
+} from "./chart_common";
+import {
+  aggregateDataForLabels,
+  filterEmptyDataPoints,
+  getChartDatasetFormat,
+  getChartDatasetValues,
+  getChartLabelValues,
+  getDefaultChartJsRuntime,
+} from "./chart_ui_common";
+
+type ComboBarChartDefinition = BarChartDefinition | ComboChartDefinition;
+
+export class ComboBarChart extends AbstractChart {
+  readonly dataSets: DataSet[];
+  readonly labelRange?: Range | undefined;
+  readonly background?: Color;
+  readonly verticalAxisPosition: VerticalAxisPosition;
+  readonly legendPosition: LegendPosition;
+  readonly stacked: boolean;
+  readonly aggregated?: boolean;
+  readonly dataSetsHaveTitle: boolean;
+
+  constructor(
+    readonly type: "bar" | "combo",
+    definition: ComboBarChartDefinition,
+    sheetId: UID,
+    getters: CoreGetters
+  ) {
+    super(definition, sheetId, getters);
+    this.dataSets = createDataSets(
+      getters,
+      definition.dataSets,
+      sheetId,
+      definition.dataSetsHaveTitle
+    );
+    this.labelRange = createRange(getters, sheetId, definition.labelRange);
+    this.background = definition.background;
+    this.verticalAxisPosition = definition.verticalAxisPosition;
+    this.legendPosition = definition.legendPosition;
+    this.stacked = definition.stacked;
+    this.aggregated = definition.aggregated;
+    this.dataSetsHaveTitle = definition.dataSetsHaveTitle;
+  }
+
+  static transformDefinition(
+    definition: ComboBarChartDefinition,
+    executed: AddColumnsRowsCommand | RemoveColumnsRowsCommand
+  ): ComboBarChartDefinition {
+    return transformChartDefinitionWithDataSetsWithZone(definition, executed);
+  }
+
+  static validateChartDefinition(
+    validator: Validator,
+    definition: ComboBarChartDefinition
+  ): CommandResult | CommandResult[] {
+    return validator.checkValidations(definition, checkDataset, checkLabelRange);
+  }
+
+  getContextCreation(): ChartCreationContext {
+    return {
+      background: this.background,
+      title: this.title,
+      range: this.dataSets.map((ds: DataSet) =>
+        this.getters.getRangeString(ds.dataRange, this.sheetId)
+      ),
+      auxiliaryRange: this.labelRange
+        ? this.getters.getRangeString(this.labelRange, this.sheetId)
+        : undefined,
+    };
+  }
+
+  copyForSheetId(sheetId: UID): ComboBarChart {
+    const dataSets = copyDataSetsWithNewSheetId(this.sheetId, sheetId, this.dataSets);
+    const labelRange = copyLabelRangeWithNewSheetId(this.sheetId, sheetId, this.labelRange);
+    const definition = this.getDefinitionWithSpecificDataSets(dataSets, labelRange, sheetId);
+    return new ComboBarChart(this.type, definition, sheetId, this.getters);
```

Comment:
That won't work. That means that when we duplicate a combo chart, the ComboChart class instance becomes a `ComboBarChart`. So we never call the constructor, lose `useBothYAxis` and call the wrong `getDefinition`.

`ComboBarChart` should be an abstract class